deploy/helm/values.yaml

```diff
@@ -118,6 +118,18 @@ trivy:
   #
   # noProxy:
 
+  # Allows to fetch the ECR Login Token if the Service is running in AWS and the EC2-VM has an Instance-Role assigned.
+  # It also needs k8s KIAM (https://github.com/uswitch/kiam) to be in place which allowed Assuming Roles to specify
+  # dedicated permissions for the Finale Role which has the rights to get login credentials.
+  #
+  # useEcrRoleCreds: false
+
+  # Set the TokenExpiry for the next AWS-Tokengeneration
+  # AWS authorization token is valid for 12 hours by default
+  # Unit is hours(h)
+  #
+  # ecrTokenRefreshTTL = 11h
```

Comment:
`ecrTokenRefreshTTL` has to be shorter then 12h?  as Trivy might want to authenticate with an expired credential

Reply:
Also, maybe a retry logic should be placed in case tokens are expired unless we force refresh before 12h

Reply:
No, Trivy will never use an expired token as in the original part I pushed, the Token creation was part of **"each"** individual request Trivy did per repository. The only reason we have this timer now in place is to reduce the number of requests (Throttling) against the ECR-API. 

So, if you want to detect an expired token and you want to make it part of a retry to refresh the token now you need to get the first trivy job triggered which tries to get the image and will run into the error.
In this case, a Token Refresh needs to be triggered. But this is a completely new scenario, which needs to be plugged into the trivy process in combination with the PlugIn routines.
Also, a drawback and a negative aspect are here is, that each request needs to be sent towards API to check if still valid or not. But this will be contradicting the Throttling which I wanted to solve.

As of now, I see no point in that we need that. The Throttling does its job, and it is up to the implementer to configure the TTLs on their own. In my view, this is the best approach.